pinot-core/src/main/java/org/apache/pinot/core/common/Operator.java

```diff
@@ -48,6 +48,9 @@
   @Nullable
   String toExplainString();
 
+  default void prepareForExplainPlan(ExplainPlanRows explainPlanRows) {
```

Comment:
Ideally we want the operator to directly fill the explain plan, and pass the plan to the child operators. Do you think it is hard to integrate this change into this PR? If so, we can add a TODO and make it handled within the operator in a separate PR

Reply:
I see. Makes sense to me. Have updated PR